api/client/build.go

```diff
@@ -209,6 +210,15 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		}
 	}
 
+	var shmSize int64 = 67108864 // initial SHM size is 64MB
```

Comment:
How about the following ? 
If 0 is specified for shm size, daemon adjusts shm size to 64MB.